benchmarks/Microsoft.AspNetCore.Routing.Performance/LinkGeneration/LinkGenerationBenchmarkBase.cs

```diff
@@ -0,0 +1,407 @@
+            endpoints[226] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/comments", "POST");
+            endpoints[227] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/comments", "GET");
+            endpoints[228] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/events", "GET");
+            endpoints[229] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/labels", "POST");
+            endpoints[230] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/labels", "GET");
+            endpoints[231] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/labels", "PUT");
+            endpoints[232] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/labels", "DELETE");
+            endpoints[233] = CreateGithubEndpoint("/repos/{owner}/{repo}/milestones/{number}/labels", "GET");
+            endpoints[234] = CreateGithubEndpoint("/repos/{owner}/{repo}/pulls/{number}/comments", "GET");
+            endpoints[235] = CreateGithubEndpoint("/repos/{owner}/{repo}/pulls/{number}/comments", "POST");
+            endpoints[236] = CreateGithubEndpoint("/repos/{owner}/{repo}/pulls/{number}/commits", "GET");
+            endpoints[237] = CreateGithubEndpoint("/repos/{owner}/{repo}/pulls/{number}/files", "GET");
+            endpoints[238] = CreateGithubEndpoint("/repos/{owner}/{repo}/pulls/{number}/merge", "PUT");
+            endpoints[239] = CreateGithubEndpoint("/repos/{owner}/{repo}/pulls/{number}/merge", "GET");
+            endpoints[240] = CreateGithubEndpoint("/repos/{owner}/{repo}/releases/{id}/assets", "GET");
+            endpoints[241] = CreateGithubEndpoint("/repos/{owner}/{repo}/issues/{number}/labels/{name}", "DELETE");
+            endpoints[242] = CreateGithubEndpoint("/repos/{owner}/{repo}/{archive_format}/{path}", "GET");
+
+            return endpoints;
+
+            MatcherEndpoint CreateGithubEndpoint(string template, string httpMethod)
+            {
+                // In attribute routing, same tempalte is used for all actions within that controller. The following
+                // simulates that where we only incremenet the controller count when a new endpoint for a new template
+                // is being created.
+                if (!templatesVisited.TryGetValue(template, out var visitedTemplateInfo))
+                {
+                    controllerCount++;
+                    visitedTemplateInfo = (controllerCount, 0);
+                    templatesVisited[template] = visitedTemplateInfo;
+                }
+
+                // Increment the action count within a controller template
+                visitedTemplateInfo.Item2++;
+
+                var requiredValues = new
+                {
+                    area = (string)null,
+                    controller = $"Controller{visitedTemplateInfo.Item1}",
+                    action = $"Action{visitedTemplateInfo.Item2}",
+                    page = (string)null
+                };
```

Comment:
I think this is a pretty good approach, and somewhat realistic as long as your considering APIs (not web sites). Nice 👍